### DISCUSSION TITLE
Calendar: Support defining custom ranges

### DISCUSSION THREAD

**Discussion by marcelkauf** [primefaces/community#2695] [PrimeNG] [answered]
I have a feature request for the calendar. We are currently using ngx-daterangepicker-material in our project. This component supports user-defined date range presets, for example you can define a preset 'last month' with timestamps for start and end. The user can then quickly select this range by clicking the corresponding button. This would be a useful enhancement for selecting data for diagrams, project management applications, etc.

Here is an example image: 
![grafik](https://github.com/user-attachments/assets/8e2f0c2c-4bdb-4e6d-a552-2dbf0a29eb35)

Live example: https://fetrarij.github.io/ngx-daterangepicker-material/custom-ranges

**Comment by qwadrox** [comment, marked as answer]
Hi! @marcelkauf I think you can do this with PrimeNG already relying on the advanced templating capabilties. 
Here is an example how I would do it: 
https://stackblitz.com/edit/xbxs5s?file=src%2Fapp%2Fcalendar-multiplemonths-demo.html
<img width="867" alt="image" src="https://github.com/user-attachments/assets/04d7521f-75d7-4a2d-bbc7-d3d9723fe8fd">

**Reply by marcelkauf** [reply]
Thank you very much for the example, we will give this a try in the next weeks.
